open-sphere-base/core/src/main/java/io/opensphere/core/util/lang/ArithmeticUtilities.java

```diff
@@ -0,0 +1,177 @@
+                || rightNumber instanceof Float)
+        {
+            returnValue = numbers.stream().collect(Collectors.summingDouble(x -> x.doubleValue()));
+        }
+        else if (leftNumber instanceof Long || rightNumber instanceof Long)
+        {
+            returnValue = numbers.stream().collect(Collectors.summingLong(x -> x.longValue()));
+        }
+        else if (leftNumber instanceof Integer || leftNumber instanceof Short || rightNumber instanceof Integer
+                || rightNumber instanceof Short)
+        {
+            returnValue = numbers.stream().collect(Collectors.summingInt(x -> x.intValue()));
+        }
+        else
+        {
+            BigDecimal sum = BigDecimal.ZERO;
+            for (Number number : numbers)
+            {
+                sum = sum.add(new BigDecimal(number.toString()));
+            }
+            returnValue = sum;
+        }
+        return returnValue;
+    }
+
+    /**
+     * Calculates the product of the supplied numbers, and returns the result.
+     * In the event of a divisor equal to zero, a null value is returned.
+     *
+     * @param dividend the number to be divided by the divisor.
+     * @param divisor the number by which the dividend is divided.
+     * @return the calculated quotient.
+     */
+    public static Number quotient(Number dividend, Number divisor)
+    {
+        if (divisor.doubleValue() == 0)
+        {
+            return null;
+        }
+        Number returnValue = null;
+        if (dividend instanceof Double || dividend instanceof Float || divisor instanceof Double || divisor instanceof Float)
+        {
+            returnValue = new Double(dividend.doubleValue() / divisor.doubleValue());
+        }
+        else if (dividend instanceof Long || divisor instanceof Long)
+        {
+            returnValue = new Long(dividend.longValue() / divisor.longValue());
+        }
+        else if (dividend instanceof Integer || divisor instanceof Integer || dividend instanceof Short
+                || divisor instanceof Short)
+        {
+            returnValue = new Integer(dividend.intValue() / divisor.intValue());
+        }
+        else
+        {
+            BigDecimal leftNumber = new BigDecimal(dividend.toString());
+            returnValue = leftNumber.divide(new BigDecimal(divisor.toString()));
+        }
+
+        return returnValue;
+    }
+
+    /**
+     * Calculates the product of the supplied numbers, and returns the result.
+     *
+     * @param left the left number portion of the product.
+     * @param right the right number portion of the product.
+     * @return the calculated product.
+     */
+    public static Number product(Number left, Number right)
+    {
+        Number returnValue = null;
+        if (left instanceof Double || left instanceof Float || right instanceof Double || right instanceof Float)
+        {
+            returnValue = new Double(left.doubleValue() * right.doubleValue() + 0.0);
```

Comment:
Why + 0.0?

Reply:
I was wondering this as well

Reply:
There's cases that Java will return a -0, due to it's implementation of IEEE Floating Point. Adding zero in this case causes the negative zero to return as positive zero.